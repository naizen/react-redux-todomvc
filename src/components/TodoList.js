import React from 'react'
import PropTypes from 'prop-types'
import TodoItem from './TodoItem'
import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd'

const TodoList = ({ filteredTodos, actions }) => {
  const onDragEnd = result => {
    if (!result.destination) {
      return
    }
    actions.reorderTodo(result)
  }

  return (
    <DragDropContext onDragEnd={onDragEnd}>
      <Droppable droppableId="droppable">
        {(provided, snapshot) => (
          <ul
            className="todo-list"
            {...provided.droppableProps}
            ref={provided.innerRef}
          >
            {filteredTodos.map((todo, index) => (
              <Draggable key={todo.id} draggableId={`${todo.id}`} index={index}>
                {(provided, snapshot) => (
                  <TodoItem
                    innerRef={provided.innerRef}
                    draggableProps={provided.draggableProps}
                    dragHandleProps={provided.dragHandleProps}
                    key={todo.id}
                    todo={todo}
                    {...actions}
                  />
                )}
              </Draggable>
            ))}
            {provided.placeholder}
          </ul>
        )}
      </Droppable>
    </DragDropContext>
  )
}

TodoList.propTypes = {
  filteredTodos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      completed: PropTypes.bool.isRequired,
      text: PropTypes.string.isRequired
    }).isRequired
  ).isRequired,
  actions: PropTypes.object.isRequired
}

export default TodoList
