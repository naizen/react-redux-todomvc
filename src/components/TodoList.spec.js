import React from 'react'
import renderer from 'react-test-renderer'
import TodoList from './TodoList'

const props = {
  filteredTodos: [
    {
      text: 'Use Redux',
      completed: false,
      id: 0
    },
    {
      text: 'Run the tests',
      completed: true,
      id: 1
    }
  ],
  actions: {
    editTodo: jest.fn(),
    deleteTodo: jest.fn(),
    completeTodo: jest.fn(),
    completeAll: jest.fn(),
    clearCompleted: jest.fn(),
    reorderTodo: jest.fn()
  }
}

describe('components', () => {
  describe('TodoList', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<TodoList {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
