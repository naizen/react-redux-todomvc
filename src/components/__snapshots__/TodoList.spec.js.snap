// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TodoList renders correctly 1`] = `
<ul
  className="todo-list"
  data-rbd-droppable-context-id="0"
  data-rbd-droppable-id="droppable"
>
  <li
    aria-labelledby="rbd-lift-instruction-0"
    className=""
    data-rbd-drag-handle-context-id="0"
    data-rbd-drag-handle-draggable-id="0"
    data-rbd-draggable-context-id="0"
    data-rbd-draggable-id="0"
    draggable={false}
    onDragStart={[Function]}
    onTransitionEnd={null}
    style={
      Object {
        "transform": null,
        "transition": null,
      }
    }
    tabIndex={0}
  >
    <div
      className="view"
    >
      <input
        checked={false}
        className="toggle"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        onDoubleClick={[Function]}
      >
        Use Redux
      </label>
      <button
        className="destroy"
        onClick={[Function]}
      />
    </div>
  </li>
  <li
    aria-labelledby="rbd-lift-instruction-0"
    className="completed"
    data-rbd-drag-handle-context-id="0"
    data-rbd-drag-handle-draggable-id="1"
    data-rbd-draggable-context-id="0"
    data-rbd-draggable-id="1"
    draggable={false}
    onDragStart={[Function]}
    onTransitionEnd={null}
    style={
      Object {
        "transform": null,
        "transition": null,
      }
    }
    tabIndex={0}
  >
    <div
      className="view"
    >
      <input
        checked={true}
        className="toggle"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        onDoubleClick={[Function]}
      >
        Run the tests
      </label>
      <button
        className="destroy"
        onClick={[Function]}
      />
    </div>
  </li>
</ul>
`;
